### PR TITLE
Allow external dependecies in custom components

### PR DESCRIPTION
**Pull request by ArcaneDiver**
Hey 👋,
with this PR i update rollup plugins to thier lastest version because plugin like `rollup-plugin-babel` have moved to rollup repo.
Moreover this resolve the problem that a custom react component can't use external dependecies. 

How i resolved the issue: 
- A new version of the [plugin](https://github.com/rollup/plugins/tree/master/packages/commonjs) that resolves commonjs modules allowed us to dont specify `namedExports` in the config
- Add this line to [resolve](https://github.com/rollup/plugins/tree/master/packages/node-resolve/) options mainFields: `['main', 'module', 'jsnext:main']`

Related issues:
#638 #650 #492

### CODE REVIEW

**Comment by wojtek-krysiak** [comment]
wow, thanks. I will chceck it out doring the weekend and merge it to beta branch

**Comment by wojtek-krysiak** [comment]
but please make sure that all 4 'yarn bundle' commands from the CI works: they are here https://github.com/SoftwareBrothers/admin-bro/blob/c5cad8d33c365a793547185d97ef8703726a61ac/.github/workflows/push.yml#L58. 

writing this because i see that in the ci there are some messages https://github.com/SoftwareBrothers/admin-bro/runs/1301890895?check_suite_focus=true

**Comment by wojtek-krysiak** [comment]
there is a bundle error in the CI (for some reason github actions passes that - but in the terminal there is some strange output - do you have the same thing on your local dev env?)

image

**Comment by ArcaneDiver** [comment]
I didn't noticed that, probably he escaped me or not. Tomorrow i will investigate.

**Comment by ArcaneDiver** [comment]
After many tests i noticed that the problem don't appear on windows (my os), then i moved to WSL which correctly throws the error. The problem is related to babel because it seem like the bundler can't recognize JSX syntax.

File: bin/watch-dev.js [hunk 1]
```diff
@@ -21,19 +21,20 @@ async function build() {
...
+        'src/backend/utils/filter/filter.ts',
+        'src/backend/decorators/**',
       ],
     },
   })
 }
 
 build().catch((error) => {
   console.log(error)
+  process.exit(1)
```
Comment (ArcaneDiver) [inline bin/watch-dev.js:39]: With this the process crash with a error code different from 0 and then make the action fail.

File: bin/watch-dev.js [hunk 2]
```diff
@@ -21,19 +21,20 @@ async function build() {
...
-        `${__dirname}/../src/utils/**`,
-        `${__dirname}/../src/backend/utils/view-helpers/view-helpers.ts`,
-        `${__dirname}/../src/backend/utils/filter/filter.ts`,
-        `${__dirname}/../src/backend/decorators/**`,
+        'src/frontend/**',
+        'src/locale/*',
+        'src/utils/**',
+        'src/backend/utils/view-helpers/view-helpers.ts',
+        'src/backend/utils/filter/filter.ts',
+        'src/backend/decorators/**',
```
Comment (ArcaneDiver) [inline bin/watch-dev.js:31]: Probably new versions of `@rollup/plugin-babel` resolves `include` paths in a different way

**Comment by ArcaneDiver** [comment]
@wojtek-krysiak finally resolved the problem, with b516e9f

**Comment by wojtek-krysiak** [comment]
wow - that is awesome - I will test it in a moment and let's merge this thing and make people happy - with their custom dependencies.

**Comment by wojtek-krysiak** [comment]
I've checked it out today and it looks fine on our test apps. But I would like to spend some time testing it on some real production application. So this is what I will do:

* this/next week we would like to publish v3.3 - we will do this without this PR
* just after that I will merge it and publish it as new beta that we will be able to test it

hope this makes sense. Once again - thanks for your work

**Review by wojtek-krysiak — APPROVED** [review]

**Pull request merged**